app/activelayer.cpp

```diff
@@ -0,0 +1,63 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "activelayer.h"
+
+ActiveLayer::ActiveLayer()
+{
+}
+
+QgsMapLayer *ActiveLayer::layer() const
+{
+  return mLayer;
+}
+
+QgsVectorLayer *ActiveLayer::vectorLayer() const
+{
+  if ( mLayer )
+    return qobject_cast<QgsVectorLayer *>( mLayer );
+
+  return nullptr;
+}
+
+QString ActiveLayer::layerId() const
+{
+  if ( mLayer )
+    return mLayer->id();
+
+  return QString();
+}
+
+QString ActiveLayer::layerName() const
+{
+  if ( mLayer )
+    return mLayer->name();
+
+  return QString();
+}
+
+void ActiveLayer::activeMapThemeChanged()
+{
+  setActiveLayer( nullptr );
```

Comment:
Instead of resetting active layer on theme change, can we check whether the previously active layer is present in the new map theme and do nothing if it is? If a user switches themes from time to time, it would be nice if the survey layer stayed the same if possible...

Reply:
sure, good idea